controllers/label_event_handler.go

```diff
@@ -26,3 +26,23 @@ func MapToOwnerByLabel(namespace, label string) func(object client.Object) []rec
 		return []reconcile.Request{}
 	}
 }
+
+// MapToControllerByMatchingLabel returns an event handler will convert events on a resource to requests
+// if the resource matches a given label key and value
+// (if the label exists)
+func MapToControllerByMatchingLabel(namespace, labelKey, labelValue string) func(object client.Object) []reconcile.Request {
+	return func(obj client.Object) []reconcile.Request {
+		if labelValueFound, exists := obj.GetLabels()[labelKey]; exists && labelValue == labelValueFound {
+			return []reconcile.Request{
+				{
+					NamespacedName: types.NamespacedName{
+						Namespace: namespace,
```

Comment:
Good point, the object namespace is fine, no need to pass that as param.

Removed in https://github.com/codeready-toolchain/toolchain-common/pull/373/commits/8a4a23dc9c9656479a50953c8b213b96a296dee6 

Thanks for pointing this out.

I've also noticed that I had to rename test.

Reply:
Don't forget that the templates can contain also cluster-scoped resources and since the controller will watch all kinds from the template, then you can get events for objects that won't have any namespace set. In other words, we need to use the operator's namespace.

Reply:
But triggering a reconcile for a Cluster Scope resource using the operator namespace ? Would this cause a not found error in the reconcile since the object needs to be searched without the namespace set ?

Reply:
never mind - I did not realize (after the PTO :smile:) that the template is actually already processed (the namespace variable is already replaced) when the controller is being started. 
All good, sorry for confusion :wink: